File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        ht = {}

        for i, num in enumerate(nums):
            diff = target - num
            if diff in ht:
                return [ht[diff], i]
            ht[num] = i
            
            

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        
        int *start = numbers.data(), *end = numbers.data() + numbers.size() - 1;

        while (start < end){
            int sum = *start + *end;
            
            if (sum == target){
                return {int(start - numbers.data() + 1), int(end - numbers.data() + 1)};
            }

            if (sum > target) end--;
            else start++;
        }
        return {};
    }
};
